File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task16_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Main {
    public static boolean solution(String str1, String str2) {
        return str1.endsWith(str2);
    }

    public static void main(String[] args) {
        System.out.println(solution("abc", "bc")); // returns true
        System.out.println(solution("abc", "d")); // returns false
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task16_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <string.h>
#include <stdio.h>

int solution(char *str1, char *str2) {
    int len1 = strlen(str1);
    int len2 = strlen(str2);
    if (len1 < len2) return 0;
    return strcmp(str1 + len1 - len2, str2) == 0;
}

int main() {
    printf("%d
", solution("abc", "bc")); // returns 1
    printf("%d
", solution("abc", "d")); // returns 0
    return 0;
}